malariagen_data/anoph/to_plink.py

```diff
@@ -0,0 +1,164 @@
+from typing import Optional
+
+import allel  # type: ignore
+import numpy as np
+import os
+import bed_reader
+
+from ..util import dask_compress_dataset
+from .snp_data import AnophelesSnpData
+from . import base_params
+from . import pca_params
+
+
+class PlinkConverter(
+    AnophelesSnpData,
+):
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        # N.B., this class is designed to work cooperatively, and
+        # so it's important that any remaining parameters are passed
+        # to the superclass constructor.
+        super().__init__(**kwargs)
+
+    def _create_plink_outfile(
+        self,
+        *,
+        output_dir,
+        region,
+        n_snps,
+        min_minor_ac,
+        thin_offset,
+        max_missing_an,
+    ):
+        return f"{output_dir}/{region}.{n_snps}.{min_minor_ac}.{thin_offset}.{max_missing_an}"
+
+    def _biallelic_snps_to_plink(
+        self,
+        *,
+        output_dir,
+        region,
+        n_snps,
+        thin_offset,
+        overwrite,
+        sample_sets,
+        sample_query,
+        sample_indices,
+        site_mask,
+        min_minor_ac,
+        max_missing_an,
+        random_seed,
+        inline_array,
+        chunks,
+    ):
```

Comment:
I'm wondering if it's worth having the code here in a separate private function, or whether actually it would be simpler to have the implementation directly within the public `biallelic_snps_to_plink()` function. There is actually nothing at all happening within the public `biallelic_snps_to_plink()` function so we are not gaining anything by splitting out the implementation into a separate private function.

Reply:
Legacy from when I copied this module from (I think) the structure of the PCA module? Amended now.